File: exp-04/q2-all-pairs-with-given-sum/solution.js
const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

let lines = [];
rl.on('line', (line) => {
    lines.push(line.trim());
    if (lines.length === 3) {
        const n = parseInt(lines[0], 10);
        const a = lines[1].split(/\s+/).map(Number);
        const s = parseInt(lines[2], 10);
        let found = false;
        for (let i = 0; i < n; i++) {
            for (let j = i + 1; j < n; j++) {
                if (a[i] + a[j] === s) {
                    console.log(`(${i}, ${j})`);
                    found = true;
                }
            }
        }
        if (!found) console.log('No pairs found');
        rl.close();
    }
});

File: exp-04/q2-all-pairs-with-given-sum/solution.py
n = int(input())
a = list(map(int, input().split()))
s = int(input())

found = False
for i in range(n):
    for j in range(i + 1, n):
        if a[i] + a[j] == s:
            print(f"({i}, {j})")
            found = True

if not found:
    print("No pairs found")